### PR TITLE
Handle source and request uuid the right way

### PR DESCRIPTION
**Pull request by kamalaknn**

### CODE REVIEW

**Comment by kunagpal** [comment]
`v6.4.0-beta.1` to go out from here once #375 is merged and this branch is up to date with develop.

File: lib/runner/replay-controller.js [hunk 1]
```diff
@@ -45,7 +46,7 @@ _.assign(ReplayController.prototype, /** @lends ReplayController.prototype */{
         !item.parent() && item.setParent(context.originalItem);
 
         // construct payload for request
-        var payload = {
+        var payload = _.merge({}, desiredPayload, {
```
Comment (kunagpal) [inline lib/runner/replay-controller.js]:
```js
_.defaults({...}, desiredPayload);
```

File: lib/runner/request-helpers-presend.js [hunk 2]
```diff
@@ -87,22 +89,27 @@ module.exports = [
...
+
+                        // pass the response to Auth `init` hook
+                        authHandler.init(authInterface, response, function (error) {
+                            if (error) { return done(error); }
+
+                            // schedule back to pre hook
+                            authPreHook();
+                        });
+                    },
+                    function (err) {
```
Comment (kunagpal) [inline lib/runner/request-helpers-presend.js]: `done` can be passed directly.

File: test/integration/request-flow/replays-spec.test.js [hunk 3]
```diff
@@ -111,6 +109,111 @@ describe('requests replayed', function () {
...
+            expect(error).to.be(null);
+
+            expect(request.url.toString()).to.eql('https://postman-echo.com/get');
+            expect(response.code).to.eql(200);
+
+            expect(trace).to.have.property('type', 'http');
+            expect(trace).to.have.property('source', 'collection');
+        });
+
+        it('should send third request as a relay', function () {
```
Comment (harryi3t) [inline test/integration/request-flow/replays-spec.test.js]: replay

**Pull request merged**